app/src/main/java/com/iic/shopingo/ui/request_flow/activities/CreateShoppingListActivity.java

```diff
@@ -27,21 +30,27 @@
   @InjectView(R.id.create_request_price_input)
   EditText priceView;
 
-  private SelectShopperActivity.SelectShopperAdapter.ShopRequest request;
+  private OutgoingRequest request;
+
+  private ShoppingList shoppingList;
 
   @Override
   protected void onCreate(Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
     request = getIntent().getParcelableExtra(EXTRAS_REQUEST_KEY);
+    if (request.getShoppingList() == null) {
+      request.setShoppingList(new ShoppingList());
+    }
+    shoppingList = request.getShoppingList();
```

Comment:
Have shoppingList be instantiated in the constructor (or in the instance variable definition).
Then you won't need this code, and won't need the setShoppingList code.